src/lsdb/dask/merge_catalog_functions.py

```diff
@@ -18,6 +19,37 @@
     from lsdb.catalog.catalog import Catalog
 
 
+def concat_partition_and_margin(
+    partition: pd.DataFrame, margin: pd.DataFrame | None, right_columns: List[str]
+) -> pd.DataFrame:
+    """Concatenates a partition and margin dataframe together
+
+    Args:
+        partition (pd.DataFrame): The partition dataframe
+        margin (pd.DataFrame): The margin dataframe
+
+    Returns:
+        The concatenated dataframe with the partition on top and the margin on the bottom
+    """
+    hive_columns = [
+        PartitionInfo.METADATA_ORDER_COLUMN_NAME,
+        PartitionInfo.METADATA_DIR_COLUMN_NAME,
+        PartitionInfo.METADATA_PIXEL_COLUMN_NAME,
+    ]
+    margin_filtered = None
+    if margin is not None:
```

Comment:
```suggestion
    if margin is not None:
      # Remove the Norder/Dir/Npix columns (used only for partitioning the margin itself), 
      # and rename the margin_Norder/Dir/Npix to take their place.
```

This is what I thought was happening in this block.